couchbase-ns/src/test/java/CouchbaseCacheLoaderMockedTest.java

```diff
@@ -0,0 +1,137 @@
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.cluster.BucketSettings;
+import com.couchbase.client.java.cluster.ClusterManager;
+import com.couchbase.client.java.env.CouchbaseEnvironment;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mache.SchemaOptions;
+import org.mache.couchbase.CouchbaseCacheLoader;
+import org.mache.couchbase.CouchbaseConfig;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.*;
+
+@PrepareForTest(CouchbaseCluster.class)
+@PowerMockIgnore("javax.management.*")
+@RunWith(PowerMockRunner.class)
+public class CouchbaseCacheLoaderMockedTest {
+
+    CouchbaseConfig config;
+    CouchbaseCacheLoader loader;
+    CouchbaseCluster mockedCluster;
+    ClusterManager mockedManager;
+
+    @Before
+    public void mockCouchbaseCluster() {
+        PowerMockito.mockStatic(CouchbaseCluster.class);
+        mockedCluster = mock(CouchbaseCluster.class);
+        when(CouchbaseCluster.create(any(CouchbaseEnvironment.class),
+                anyListOf(String.class))).thenReturn(mockedCluster);
+        mockedManager = mock(ClusterManager.class);
+        when(mockedCluster.clusterManager(anyString(), anyString())).thenReturn(mockedManager);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowIllegalArgumentWhenNotGivenConfig() throws Exception {
+        new CouchbaseCacheLoader<>(null);
+    }
+
+    @Test
+    public void shouldCreateConnection() throws Exception {
+        givenCacheLoaderWith(SchemaOptions.USEEXISTINGSCHEMA);
+        loader.create();
+        thenClusterManagerAndBucketCreated(mockedCluster);
+    }
+
+    @Test
+    public void shouldCreateBucket() throws Exception {
+        givenCacheLoaderWith(SchemaOptions.CREATESCHEMAIFNEEDED);
+
+        when(mockedManager.hasBucket(anyString())).thenReturn(false);
+        when(mockedCluster.clusterManager(anyString(), anyString())).thenReturn(mockedManager);
+
+        loader.create();
+
+        thenBucketCreated();
+        thenClusterManagerAndBucketCreated(mockedCluster);
+    }
+
+    @Test
+    public void shouldDropAndCreateBucket() throws Exception {
+        givenCacheLoaderWith(SchemaOptions.CREATEANDDROPSCHEMA);
+
+        when(mockedManager.hasBucket(anyString())).thenAnswer(new Answer<Boolean>() {
+            boolean returnValue = false;

```

Comment:
Consider extracting the answer into a named thing, to aid readability of the test